google/cloud/forseti/services/inventory/storage.py

```diff
@@ -181,6 +182,20 @@ class Inventory(BASE):
     other = Column(Text)
     inventory_errors = Column(Text)
 
+    __table_args__ = (
+        Index('idx_category_resource',
```

Comment:
I am not a db/indexing expert, can you please help me to understand why we are creating the 3 indexes this way?  Thanks!

Reply:
The indexes are built based on the filters from https://github.com/GoogleCloudPlatform/forseti-security/blob/2.0-dev/google/cloud/forseti/services/inventory/storage.py#L790.

idx_category_resource handles the primary filter:

WHERE inventory_index_id = self.inventory_index.id AND category=<category> AND resource_type IN (<resource_types>)

idx_resource_type_id and idx_parent_resource_type_id are specifically for the with_parent join:

https://github.com/GoogleCloudPlatform/forseti-security/blob/2.0-dev/google/cloud/forseti/services/inventory/storage.py#L852

Reply:
Also, idx_resource_type_id and idx_parent_resource_type_id is for get_root().